hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetRecordHandler.java

```diff
@@ -58,13 +70,73 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var op = context.query().transactionGetRecordOrThrow();
```

Comment:
Do we need to check if context has a query or if `context.query()` is null before this, or is the NPE fine here?

Reply:
I think you told me if the first 3 lines of the code NPE is ok

Reply:
base on the code I checked query cannot be null
```
   public QueryContextImpl(
            @NonNull final ReadableStoreFactory storeFactory,
            @NonNull final Query query,
            @NonNull final Configuration configuration) {
        this.storeFactory = requireNonNull(storeFactory, "The supplied argument 'storeFactory' cannot be null!");
        this.query = requireNonNull(query, "The supplied argument 'query' cannot be null!");
        this.configuration = requireNonNull(configuration, "The supplied argument 'configuration' cannot be null!");
    }
```

Reply:
The NPE from `transactionGetRecordOrThrow()` is possibly more concerning, but I'll leave it to you if we should handle that differently.